src/main/java/org/spongepowered/api/item/data/map/MapCanvas.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.item.data.map;
+
+import com.google.common.base.Optional;
+
+import java.awt.Color;
+import java.awt.Image;
+
+/**
+ * A class that represents the display of a map item.
+ */
+public interface MapCanvas {
+    /**
+     * Gets the {@link MapData} that this canvas is attached to.
+     *
+     * @return The {@link MapData} for this canvas
+     */
+    MapData getMapData();
+
+    /**
+     * Gets the {@link MapColor} at the specified pixel position. If the position
+     * is invalid (out of range), the value will be absent. Pixels start at (0,0)
```

Comment:
There should also be a method `Vector2i getSize()` for this purpose.
Otherwise you dont know which pixels you can access. Because mods may be able to change the LOD for maps as well.